### DISCUSSION TITLE
Documentation on adding a new rule

### DISCUSSION THREAD

**Discussion by marschwar** [detekt/detekt#3549] [Q&A] [answered]
The contribution guide states:

> ... do not forget to run ./gradlew build. This will execute tests locally and update the default-detekt.config.yml as well as add the new/changed rules to the documentation.

The problem seems to be that `./gradlew build` fails after adding a new rule since the `DetektYmlConfigSpec` checks that the `default-detekt-config.yml` contains the correct number of rules. It seems to be run before the task `detekt-generator:generateDocumentation`. Am I doing something wrong here or is the documentation outdated or is the build misconfigured?

**Comment by BraisGabin** [comment, marked as answer]
The documentation is outdated. I think that we should update the docs and say run: `generateDocumentation`.

**Reply by marschwar** [reply]
Thank you for clarification. 

Is there a way to mark a discussion as closed? I get the feeling, I should have opened an issue instead.

**Reply by cortinico** [reply]
Done :) That was actually a good candidate for a discussion. You should have opened in the Q&A section that allows to mark them as "answered"
